joomla/joomla-cms
# [5.4] Carry through disabled attribute to sub-fields in media and subform custom fields

Pull Request for Issue #38600 .

### Summary of Changes
Added some checks in the plugins and layout fields to see if the parent fields are set as disabled. If so, set subfields to disabled also.

~~This is IN PROGRESS, feedback welcome.~~
**Ready for review!**

~~Left to do:~~ Done!
- [x] Something funny is happening with disabled subforms that have media fields when you save them.
- [x] Make sure buttons on repeatable subforms are disabled or hidden when subform is disabled.

### Testing Instructions

- Make sure you have two users with different user groups, both with edit access to articles
- As superuser, create a custom field of type "media".
  - In the "options" tab, set "display when read only" to "yes"
  - In the "permissions" tab, set the edit permissions for your other user to denied.
- As superuser, create a custom field of type "subform" with at least one subfield.
  - In the "options" tab, set "display when read only" to "yes"
  - In the "permissions" tab, set the edit permissions for your other user to denied.
- Login as the other user and navigate to the edit form where the custom fields are visible.

### Actual result BEFORE applying this Pull Request

Media and subform fields are visible AND can be edited, or appear to be editable. Rows can be added and removed. Fields can be changed and text can be entered. However, any changes will not be saved to the database.

### Expected result AFTER applying this Pull Request

Media and subform fields are visible but cannot be edited. Fields for media select, alt text, etc are disabled. Rows cannot be added or removed.



### Documentation Changes Required
None to my knowledge.
